eclipse-collections/src/main/java/org/eclipse/collections/impl/set/immutable/AbstractImmutableSet.java

```diff
@@ -472,6 +473,14 @@ public boolean isProperSubsetOf(SetIterable<? extends T> candidateSuperset)
     @Override
     public ParallelUnsortedSetIterable<T> asParallel(ExecutorService executorService, int batchSize)
     {
-        return this.toSet().asParallel(executorService, batchSize);
+        if (executorService == null)
+        {
+            throw new NullPointerException();
+        }
+        if (batchSize < 1)
+        {
+            throw new IllegalArgumentException();
```

Comment:
The IllegalArgumentException should have some context, like the batchSize passed in.